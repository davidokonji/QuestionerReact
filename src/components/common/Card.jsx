import React from 'react';
import moment from 'moment';
import { Link } from 'react-router-dom';

const Card = ({
  id, images, title, location, tags, happeningOn
}) => {
  const month = moment(happeningOn).format('MMM');
  const day = moment(happeningOn).format('D');
  const date = moment(happeningOn).format('dddd, MMMM Do YYYY');
  return (
    <div
      className='card col-md-3 col-lg-3 my-3 shadow-sm p-0 mx-3'
      role='button'
      tabIndex='-1'
      onKeyDown={null}
    >
      <div className='card-body p-0'>
        <div
          className='card-img'
        >
          <img src={images} alt='' className='card-img-top card-img' />
        </div>
        <div className='row col-12 h-auto'>
          <div className='col-2 col-md-2 d-flex flex-column side-card p-0'>
            <span className='mt-5 side-card-value mx-auto'>
              {month}
            </span>
            <span className='mt-2 side-card-value mx-auto'>
              {day}
            </span>
          </div>
          <div className='col-10 col-md-10'>
            <div className='col-12 my-3'>
              <Link to={`/meetup/${id}`} className='stretched-link card-link'>
                {title}
              </Link>
            </div>
            <div className='col-12 my-3'>
              {date}
            </div>
            <div className='col-12 my-3'>
              {location}
            </div>
            <div className='col-12 my-3'>
              <span className=''>
                {tags}
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Card;
