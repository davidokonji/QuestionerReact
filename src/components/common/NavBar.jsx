import React from 'react';
import { NavLink, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import {
  Collapse,
  Navbar,
  NavbarToggler,
  Nav,
  NavItem,
  UncontrolledDropdown,
  DropdownToggle,
  DropdownMenu,
  DropdownItem
} from 'reactstrap';

import { logoutUser } from '../../actions';
import { user } from '../../Assets';

class NavBar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isOpen: false
    };
  }

  toggle = () => {
    const { isOpen } = this.state;
    this.setState({
      isOpen: !isOpen
    });
  }

  logout = () => {
    const { Logout, history: { push } } = this.props;
    Logout();
    push('/');
  }

  render() {
    const { isOpen } = this.state;
    const {
      isAdmin,
      history: { push },
      profile
    } = this.props;
    return (
      <div>
        <Navbar expand='md' dark className='p-3 nav-bar'>
          <NavLink to='/dashboard' className='nav-title'>
              Questioner
          </NavLink>
          <NavbarToggler onClick={this.toggle} className='toggle' />
          <Collapse isOpen={isOpen} navbar>
            <Nav className='ml-auto' navbar>
              {
              isAdmin && (
                <NavItem>
                  <NavLink to='/admin' className='nav-link'>Admin</NavLink>
                </NavItem>
              )
            }
              <NavItem>
                <NavLink to='/dashboard' className='nav-link'>Dashboard</NavLink>
              </NavItem>
              <UncontrolledDropdown nav inNavbar>
                <DropdownToggle nav caret>
                  <span>
                    <img
                      src={
                        profile === undefined
                        || profile.images === null
                          ? user
                          : profile.images[0]}
                      alt='user'
                      style={{ width: '3.5rem', height: '3.5rem', objectFit: 'contain' }}
                      className='rounded-circle'
                    />
                  </span>
                </DropdownToggle>
                <DropdownMenu right className='p-3'>
                  <DropdownItem onClick={() => push('/user')}>
                        Your profile
                  </DropdownItem>
                  <DropdownItem divider />
                  <DropdownItem onClick={this.logout} className='logout'>
                       Sign Out
                  </DropdownItem>
                </DropdownMenu>
              </UncontrolledDropdown>
            </Nav>
          </Collapse>
        </Navbar>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  isAdmin: state.auth.isAdmin,
  profile: state.profile.user,
});

const mapDispatchToProps = {
  Logout: logoutUser
};

const NavBarConnected = withRouter(connect(mapStateToProps, mapDispatchToProps)(NavBar));
export {
  NavBar,
  NavBarConnected
};
