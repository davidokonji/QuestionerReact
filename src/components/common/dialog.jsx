import React from 'react';
import {
  Button,
  Modal,
  ModalHeader,
  ModalBody,
  ModalFooter
} from 'reactstrap';

class DeleteDialog extends React.Component {
  render() {
    const { open, className, toggle } = this.props;
    return (
      <div>
        <Modal isOpen={open} toggle={toggle} className={className}>
          <ModalHeader toggle={this.toggle}>Modal title</ModalHeader>
          <ModalBody>
            About to delete
          </ModalBody>
          <ModalFooter>
            <Button color='primary' onClick={toggle}>Delete</Button>
            <Button color='secondary' onClick={toggle}>Cancel</Button>
          </ModalFooter>
        </Modal>
      </div>
    );
  }
}

export default DeleteDialog;
