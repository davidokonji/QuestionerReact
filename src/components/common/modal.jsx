import React, { Component } from 'react';
import {
  Button, Modal, ModalHeader, ModalBody, ModalFooter
} from 'reactstrap';

class ModalComponent extends Component {
  render() {
    const {
      open,
      toggle,
      className,
      children
    } = this.props;
    const externalCloseBtn = (
      <button
        className='close'
        type='button'
        style={{ position: 'absolute', top: '15px', right: '15px' }}
        onClick={toggle}
      >
        &times;
      </button>
    );

    return (
      <div>
        <Modal
          isOpen={open}
          toggle={toggle}
          className={className}
          external={externalCloseBtn}
        >
          <ModalBody>
            {children}
          </ModalBody>
        </Modal>
      </div>
    );
  }
}

export default ModalComponent;
