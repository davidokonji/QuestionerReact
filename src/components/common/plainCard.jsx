import React from 'react';

const PlainCard = ({
  id,
  topic,
  location,
  date,
  history
}) => (
  <div
    className='col-12 col-md-4 mt-3 plaincard'
    onClick={() => history.push(`/meetup/${id}`)}
    role='button'
    tabIndex='-1'
    onKeyDown={null}
  >
    <div className='card shadow-sm'>
      <div className='card-body col-12'>
        <h3 className='mt-0'>{topic}</h3>
        <p className='mt-0'>{location}</p>
        <p className='mt-0'>{date}</p>
      </div>
    </div>
  </div>
);

export default PlainCard;
