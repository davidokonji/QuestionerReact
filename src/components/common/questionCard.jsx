import React from 'react';
import { upArrow, downArrow } from '../../Assets';

const Questions = () => (
  <div className='py-3'>
    <div className='card col-12 col-md-10 shadow-sm'>
      <div className='card-body p-0 row'>
        <div className='col-1 col-md-1 d-flex flex-column justify-content-center my-2'>
          <p className='col'>
            <img src={upArrow} alt='uparrow' className='h-25 arrow-icon' />
          </p>
          <p className='col mx-3'>
                0
          </p>
          <p className='col'>
            <img src={downArrow} alt='downarrow' className='h-25 arrow-icon' />
          </p>
        </div>
        <div className='col-9 col-md-9 row'>
         question
        </div>
      </div>
    </div>
  </div>
);

export default Questions;
