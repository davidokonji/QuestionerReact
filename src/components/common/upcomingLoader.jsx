import React from 'react';
import ContentLoader from 'react-content-loader';

const UpcomingLoader = () => (
  <div className='col-4 col-md-4'>
    <ContentLoader
      height={160}
      width={350}
      speed={2}
      primaryColor='#f3f3f3'
      secondaryColor='#ecebeb'
    >
      <rect x='184' y='123' rx='0' ry='0' width='0' height='0' />
      <rect x='247' y='60' rx='0' ry='0' width='0' height='0' />
      <rect x='162' y='80' rx='0' ry='0' width='0' height='0' />
      <rect x='28' y='15' rx='0' ry='0' width='249' height='18' />
      <rect x='221' y='60' rx='0' ry='0' width='0' height='0' />
      <rect x='28' y='50' rx='0' ry='0' width='249' height='12' />
      <rect x='27' y='97' rx='0' ry='0' width='249' height='14' />
      <rect x='28' y='75' rx='0' ry='0' width='187' height='11' />
    </ContentLoader>
  </div>
);

export default UpcomingLoader;
