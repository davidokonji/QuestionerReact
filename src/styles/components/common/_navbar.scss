.nav-bar {
    background-color: $dark-blue;
    text-transform: uppercase;
    .nav-title {
        color: $white;
        font: {
            family: 'Montserrat', sans-serif;
            size: 2rem;
        }
        text-transform: uppercase;
        text-decoration: none;
    }
    .nav-link{
        color: $white;
    }
}