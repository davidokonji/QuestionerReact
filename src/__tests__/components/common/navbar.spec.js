import { shallow } from 'enzyme';
import React from 'react';
import { NavBar } from '../../../components/common/NavBar';

describe('Nav bar component', () => {
  it('should render nav bar correctly', () => {
    const wrapper = shallow(<NavBar isAdmin />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render nav bar correctly', () => {
    const Logout = jest.fn();
    const wrapper = shallow(<NavBar Logout={Logout} isAdmin={false} />);

    wrapper.setState({
      isOpen: false
    });
    wrapper.find('.toggle').simulate('click');
    wrapper.find('.logout').simulate('click');
    expect(wrapper).toMatchSnapshot();
  });
});
