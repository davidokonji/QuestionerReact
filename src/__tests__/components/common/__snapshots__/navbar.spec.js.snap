// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav bar component should render nav bar correctly 1`] = `
<div>
  <Navbar
    className="p-3 nav-bar"
    dark={true}
    expand="md"
    tag="nav"
  >
    <NavLink
      className="nav-title"
      to="/dashboard"
    >
      Questioner
    </NavLink>
    <NavbarToggler
      className="toggle"
      onClick={[Function]}
      tag="button"
      type="button"
    />
    <Collapse
      appear={false}
      enter={true}
      exit={true}
      in={false}
      isOpen={false}
      mountOnEnter={false}
      navbar={true}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      tag="div"
      timeout={350}
      unmountOnExit={false}
    >
      <Nav
        className="ml-auto"
        navbar={true}
        tag="ul"
        vertical={false}
      >
        <NavItem
          tag="li"
        >
          <NavLink
            className="nav-link"
            to="/admin"
          >
            Admin
          </NavLink>
        </NavItem>
        <NavItem
          tag="li"
        >
          <NavLink
            className="nav-link"
            to="/dashboard"
          >
            Dashboard
          </NavLink>
        </NavItem>
        <UncontrolledDropdown
          inNavbar={true}
          nav={true}
        >
          <DropdownToggle
            aria-haspopup={true}
            caret={true}
            color="secondary"
            nav={true}
          >
            <span>
              <img
                alt="user"
                className="rounded-circle"
                src={
                  Object {
                    "canInstrument": true,
                    "createTransformer": [Function],
                    "getCacheKey": [Function],
                    "process": [Function],
                  }
                }
                style={
                  Object {
                    "height": "3.5rem",
                    "objectFit": "contain",
                    "width": "3.5rem",
                  }
                }
              />
            </span>
          </DropdownToggle>
          <DropdownMenu
            className="p-3"
            flip={true}
            right={true}
            tag="div"
          >
            <DropdownItem
              onClick={[Function]}
              tag="button"
              toggle={true}
            >
              Your profile
            </DropdownItem>
            <DropdownItem
              divider={true}
              tag="button"
              toggle={true}
            />
            <DropdownItem
              className="logout"
              onClick={[Function]}
              tag="button"
              toggle={true}
            >
              Sign Out
            </DropdownItem>
          </DropdownMenu>
        </UncontrolledDropdown>
      </Nav>
    </Collapse>
  </Navbar>
</div>
`;

exports[`Nav bar component should render nav bar correctly 2`] = `
<div>
  <Navbar
    className="p-3 nav-bar"
    dark={true}
    expand="md"
    tag="nav"
  >
    <NavLink
      className="nav-title"
      to="/dashboard"
    >
      Questioner
    </NavLink>
    <NavbarToggler
      className="toggle"
      onClick={[Function]}
      tag="button"
      type="button"
    />
    <Collapse
      appear={false}
      enter={true}
      exit={true}
      in={false}
      isOpen={true}
      mountOnEnter={false}
      navbar={true}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      tag="div"
      timeout={350}
      unmountOnExit={false}
    >
      <Nav
        className="ml-auto"
        navbar={true}
        tag="ul"
        vertical={false}
      >
        <NavItem
          tag="li"
        >
          <NavLink
            className="nav-link"
            to="/dashboard"
          >
            Dashboard
          </NavLink>
        </NavItem>
        <UncontrolledDropdown
          inNavbar={true}
          nav={true}
        >
          <DropdownToggle
            aria-haspopup={true}
            caret={true}
            color="secondary"
            nav={true}
          >
            <span>
              <img
                alt="user"
                className="rounded-circle"
                src={
                  Object {
                    "canInstrument": true,
                    "createTransformer": [Function],
                    "getCacheKey": [Function],
                    "process": [Function],
                  }
                }
                style={
                  Object {
                    "height": "3.5rem",
                    "objectFit": "contain",
                    "width": "3.5rem",
                  }
                }
              />
            </span>
          </DropdownToggle>
          <DropdownMenu
            className="p-3"
            flip={true}
            right={true}
            tag="div"
          >
            <DropdownItem
              onClick={[Function]}
              tag="button"
              toggle={true}
            >
              Your profile
            </DropdownItem>
            <DropdownItem
              divider={true}
              tag="button"
              toggle={true}
            />
            <DropdownItem
              className="logout"
              onClick={[Function]}
              tag="button"
              toggle={true}
            >
              Sign Out
            </DropdownItem>
          </DropdownMenu>
        </UncontrolledDropdown>
      </Nav>
    </Collapse>
  </Navbar>
</div>
`;
